packages/perseus/src/widgets/interactive-graphs/graphs/linear.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from "react";
+
+import {moveControlPoint, moveSegment} from "../interactive-graph-action";
+
+import {MovableLine} from "./components/movable-line";
+import {StyledMovablePoint} from "./components/movable-point";
+
+import type {InteractiveLineProps} from "./types";
+import type {MafsGraphProps, LinearGraphState} from "../types";
+import type {vec} from "mafs";
+
+type LinearGraphProps = MafsGraphProps<LinearGraphState>;
+
+export const LinearGraph = (props: LinearGraphProps) => {
+    const {dispatch} = props;
+    const {coords: lines, snapStep, range, type} = props.graphState;
+
+    const colors = ["var(--movable-line-stroke-color)", "var(--mafs-violet)"];
+
+    return (
+        <>
+            {lines?.map((line, i) => (
+                <LineView
+                    key={i}
```

Comment:
**question**: 🤔  I wonder if there's any "stronger" line identifier than its index?  or if we should give each line some type of stable ID?

Reply:
I don't think there is at the moment. However, the index for these lines will not change for the user. Definitely something we will need to consider for more dynamic graphs like the Unlimited Points.